src/config/functions.ts

```diff
@@ -32,6 +34,22 @@ export const fnSchema = {
 	osVariant: () => {
 		return osRelease.getOSVariant(constants.hostOSVersionPath);
 	},
+	deviceArch: async () => {
+		try {
+			// FIXME: We should be mounting the following file into the supervisor from the
+			// start-resin-supervisor script, changed in meta-resin - but until then, hardcode it
+			const data = await fs.readFile(
+				`${constants.rootMountPoint}/resin-boot/device-type.json`,
+				'utf8',
+			);
+			const deviceInfo = JSON.parse(data);
+
+			return deviceInfo.arch;
+		} catch (e) {
+			log.error(`Unable to get architecture: ${e}`);
+			return 'unavailable';
```

Comment:
Why not `unknown` as in `schema-type.ts` ?

Reply:
I was originally thinking it was for different failure modes, but it's not actually possible with the current situation for `unavailable` to be returned. I guess it's fine.